### PR TITLE
PRP-202: Finish /name

### PR DESCRIPTION
**Pull request by makaelastephens**
## Ticket

https://wicmtdp.atlassian.net/browse/PRP-202

## Changes
- Fixed the error validation so the proper error message shows up
- Routes to /count
- Added e2e tests

## Context for reviewers
- I DM'd with Sasha about this - I wasn't sure how to pass in the field IDs for the name fields, so I did a workaround...

## Testing
image

### CODE REVIEW

File: participant/app/routes/$localAgency/recertify/name.tsx [hunk 1]
```diff
@@ -4,13 +4,68 @@ import { Trans } from "react-i18next";
...
+  const parsedForm = representativeNameSchema.parse(formData);
+  const { submissionID } = await cookieParser(request);
+  console.log(`Got submission ${JSON.stringify(parsedForm)}`);
+  await upsertSubmissionForm(submissionID, "name", parsedForm);
+  const routeTarget = routeFromName(request);
+  console.log(`Completed name form; routing to ${routeTarget}`);
+  return redirect(routeTarget);
+};
+
+export default function Namex() {
```
Comment (rocketnova) [inline participant/app/routes/$localAgency/recertify/name.tsx]: Is this `Namex()` instead of `Name()` because `Name()` is a reserved function name?

**Review by rocketnova — APPROVED** [review]
@makaelastephens This looks great to me! :shipit: No changes requested. 

Regarding:

> I DM'd with Sasha about this - I wasn't sure how to pass in the field IDs for the name fields, so I did a workaround...

I see what you mean about using the `nameSchemaFactory()`, but I think what you did was clever and makes sense.

```
export const representativeNameSchema = nameSchemaFactory("representative");
```

File: participant/app/routes/$localAgency/recertify/name.tsx (hunk 1 above)
Comment (makaelastephens) [reply participant/app/routes/$localAgency/recertify/name.tsx]: Potentially it's a typo... I just noticed it said `Index()` which didn't seem right. Are you suggesting `Name()` wouldn't work?

Comment (rocketnova) [reply participant/app/routes/$localAgency/recertify/name.tsx]: No, I was not suggesting `Name()` wouldn't work. I was just curious why it was `Namex()` and speculating

**Pull request merged**